src/main/org/deidentifier/arx/metric/v2/MetricMDNUEntropyPrecomputed.java

```diff
@@ -326,7 +341,73 @@ protected void initializeInternal(final DataManager manager,
             max[i] = (input.getDataLength() * log2(input.getDataLength())) * Math.max(gFactor, sFactor);
         }
         
+        if (config.isPrivacyModelSpecified(DataDependentEDDifferentialPrivacy.class)) {
+            
+            // Store minimal size of equivalence classes
+            DataDependentEDDifferentialPrivacy dpCriterion = config.getPrivacyModel(DataDependentEDDifferentialPrivacy.class);
+            k = (double)dpCriterion.getK();
```

Comment:
The minimal class size is extracted at multiple locations, so this comment should be considered wherever applicable:

Generic access to minimal class-size

        // For each
        for (PrivacyCriterion c : this.getPrivacyModels()) {
            if (c.isMinimalClassSizeAvailable()) {
                result = Math.max(result, c.getMinimalClassSize());
            }
        }

In this case: dpCriterion.getMinimalClassSize();

Reply:
In the same context: please also overwrite getRiskThresholdJournalist(), getRiskThresholdMarkter() and getRiskThresholdProsecutor() in AbstractDPCriterion and return 1/k in each case.

Reply:
Unrelated issue: can we rename the following classes please (if this does not introduce serializability issues):
AbstractEDDifferentialPrivacy -> EDDifferentialPrivacyAbstract
DataDependentEDDifferentialPrivacy -> EDDifferentialPrivacyOptimal